From e4e87d99fde943d6b4341beb91ed96b7c0cb589a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:59 -0400
Subject: [tools] perf/evlist: Fix max mmap_pages

Message-id: <1408953210-26343-112-git-send-email-jolsa@redhat.com>
Patchwork-id: 88824
O-Subject: [PATCH RHEL7.1 BZ1133083 111/322] perf evlist: Fix max mmap_pages
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f5ae9c424e4c80c44e9ab7e5ff4a6b79490c23be
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Mon Dec 9 15:18:38 2013 +0200

description
===========
'SIZE_MAX / page_size' is an upper limit for the maximum number of mmap
pages, not a lower limit.  Change the condition accordingly.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evlist.c b/tools/perf/util/evlist.c
index 4d0945c..98ec96b 100644
--- a/tools/perf/util/evlist.c
+++ b/tools/perf/util/evlist.c
@@ -754,7 +754,7 @@ int perf_evlist__parse_mmap_pages(const struct option *opt, const char *str,
  unsigned long max = UINT_MAX;
  long pages;
 
- if (max < SIZE_MAX / page_size)
+ if (max > SIZE_MAX / page_size)
   max = SIZE_MAX / page_size;
 
  pages = parse_pages_arg(str, 1, max);
-- 
1.7.1